src/helpers/similarity.ts

```diff
@@ -0,0 +1,86 @@
+import { getBotContext } from "../bindings";
+import { Payload, Choices } from "../types";
+import { listAllIssuesForRepo, upsertCommentToIssue } from "../helpers/issue";
+import axios from "axios";
+
+export const findDuplicateOne = async () => {
+  const { payload: _payload } = getBotContext();
+  const issue = (_payload as Payload).issue;
+  const issues = await listAllIssuesForRepo("all");
+  issues.shift();
+  const importantWords = await extractImportantWords();
+  const wordCount = importantWords.length;
+  for (const iss of issues) {
+    if (iss.body) {
+      const probability = (countIncludedWords(iss.body, importantWords) * 100.0) / wordCount;
+      if (probability > parseInt(process.env.SIMILARITY_THRESHOLD || "80")) {
+        if (issue?.number) {
+          await upsertCommentToIssue(issue?.number, `Similar issue (${iss.title}) found at ${iss.html_url}.\nSimilarity is about ${probability}%`, "created");
+          return;
+        }
+      }
+    }
+  }
+};
+
+const countIncludedWords = (inputString: string, words: string[]): number => {
+  const lowerCaseString = inputString.toLowerCase(); // Convert the string to lowercase for case-insensitive matching
+
+  let count = 0;
+  for (const word of words) {
+    if (lowerCaseString.includes(word.toLowerCase())) {
```

Comment:
Yes, indexOf is a a tiny, tiny amount faster than includes but in reality this will make no difference. Anyway, I will change this.

Reply:
Perhaps it makes sense to focus on code readability vs optimizing for marginal performance benefits (I think this would make more sense if we were looping in the 100,000+ times range. Perhaps we can focus on performance optimizations as needed. In the meantime lets maintain `includes` for readability reasons.